lib/Conversion/VerifToSMT/VerifToSMT.cpp

```diff
@@ -150,21 +159,41 @@ struct LogicEquivalenceCheckingOpConversion
     rewriter.create<smt::AssertOp>(loc, toAssert);
 
     // Fifth, check for satisfiablility and report the result back.
-    Value falseVal =
-        rewriter.create<arith::ConstantOp>(loc, rewriter.getBoolAttr(false));
-    Value trueVal =
-        rewriter.create<arith::ConstantOp>(loc, rewriter.getBoolAttr(true));
-    auto checkOp = rewriter.create<smt::CheckOp>(loc, rewriter.getI1Type());
-    rewriter.createBlock(&checkOp.getSatRegion());
-    rewriter.create<smt::YieldOp>(loc, falseVal);
-    rewriter.createBlock(&checkOp.getUnknownRegion());
-    rewriter.create<smt::YieldOp>(loc, falseVal);
-    rewriter.createBlock(&checkOp.getUnsatRegion());
-    rewriter.create<smt::YieldOp>(loc, trueVal);
-    rewriter.setInsertionPointAfter(checkOp);
-    rewriter.create<smt::YieldOp>(loc, checkOp->getResults());
-
-    rewriter.replaceOp(op, solver->getResults());
+    // If no operation uses the result of this solver, we leave our check
+    // operations empty. If the result is used, we create a check operation with
+    // the result type of the operation and yield the result of the check
+    // operation.
+    if (unusedResult) {
+      auto checkOp = rewriter.create<smt::CheckOp>(loc, TypeRange{});
+      rewriter.createBlock(&checkOp.getSatRegion());
+      rewriter.create<smt::YieldOp>(loc);
+      rewriter.createBlock(&checkOp.getUnknownRegion());
+      rewriter.create<smt::YieldOp>(loc);
+      rewriter.createBlock(&checkOp.getUnsatRegion());
+      rewriter.create<smt::YieldOp>(loc);
+      rewriter.setInsertionPointAfter(checkOp);
+      rewriter.create<smt::YieldOp>(loc);
+
+      // Erase as operation is replaced by an operator without a return value.
+      rewriter.eraseOp(op);
+    } else {
+      Value falseVal =
+          rewriter.create<arith::ConstantOp>(loc, rewriter.getBoolAttr(false));
+      Value trueVal =
+          rewriter.create<arith::ConstantOp>(loc, rewriter.getBoolAttr(true));
+      auto checkOp = rewriter.create<smt::CheckOp>(loc, rewriter.getI1Type());
+      rewriter.createBlock(&checkOp.getSatRegion());
+      rewriter.create<smt::YieldOp>(loc, falseVal);
+      rewriter.createBlock(&checkOp.getUnknownRegion());
+      rewriter.create<smt::YieldOp>(loc, falseVal);
+      rewriter.createBlock(&checkOp.getUnsatRegion());
+      rewriter.create<smt::YieldOp>(loc, trueVal);
+      rewriter.setInsertionPointAfter(checkOp);
+      rewriter.create<smt::YieldOp>(loc, checkOp->getResults());
+
+      rewriter.replaceOp(op, solver->getResults());
+    }
+
```

Comment:
It seems like there's no unit test for the changes in this file (only the integration test, thus not super critical but still good to have).